From Dan Gohman <dev@sunfishcode.online>
Subject: [PATCH] Fix the flags argument of `pidfd_open`.

This corrects the flags value of `pidfd_open` to avoid passing
target flags to the host. Currently the only flag is `PIDFD_NONBLOCK`
so we use the `fcntl_flags_tbl` to translate it.

--- a/linux-user/syscall.c
+++ b/linux-user/syscall.c
@@ -9477,7 +9477,8 @@
 #endif
 #if defined(__NR_pidfd_open) && defined(TARGET_NR_pidfd_open)
     case TARGET_NR_pidfd_open:
-        return get_errno(pidfd_open(arg1, arg2));
+        return get_errno(pidfd_open(arg1,
+			    target_to_host_bitmask(arg2, fcntl_flags_tbl)));
 #endif
 #if defined(__NR_pidfd_send_signal) && defined(TARGET_NR_pidfd_send_signal)
     case TARGET_NR_pidfd_send_signal:
